### PR TITLE
Copter: Payload Place: Improve touchdown test

### PR DESCRIPTION
**Pull request by lthall**
This PR helps address a problem where precision place can detect touch down prematurely. The existing code based the hover throttle and decent throttle based on a single reading. This PR replaces this with the minimum reading over 1 second.

### CODE REVIEW

File: ArduCopter/mode_auto.cpp [hunk 1]
```diff
@@ -1917,61 +1918,58 @@ bool ModeAuto::verify_payload_place()
...
         }
-        // watch the throttle to determine whether the load has been placed
-        // debug("hover ratio: %f   descend ratio: %f\n", current_throttle_level/nav_payload_place.hover_throttle_level, ((nav_payload_place.descend_throttle_level == 0) ? -1.0f : current_throttle_level/nav_payload_place.descend_throttle_level));
-        if (current_throttle_level/nav_payload_place.hover_throttle_level > hover_throttle_placed_fraction &&
-            (is_zero(nav_payload_place.descend_throttle_level) ||
-             current_throttle_level/nav_payload_place.descend_throttle_level > descent_throttle_placed_fraction)) {
+        // check for reduced thrust requirement indicating possible payload touch down
+        if (current_throttle_level / nav_payload_place.hover_throttle_level > hover_throttle_placed_fraction &&
+            (now - nav_payload_place.descend_start_timestamp > descend_throttle_calibrate_time ||
+             current_throttle_level / nav_payload_place.descend_throttle_level > descent_throttle_placed_fraction)) {
```
Comment (IamPete1) [inline ArduCopter/mode_auto.cpp:1966]: Existing issue, but more likely now. `hover_throttle_level` and `descend_throttle_level` could be zero and give us a FPE.

**Review by IamPete1 — COMMENTED** [review]
Basically looks OK, I do wonder if the min might leave us venerable to using two low a throttle especially if the Z controller tune is quite aggressive. Maybe a average over the one second would be better?

This feature is rarely used, if this has had some testing I would be inclined to merge.

File: ArduCopter/mode_auto.cpp (hunk 1 above)
Comment (lthall) [reply ArduCopter/mode_auto.cpp]: I have moved the reference thrust level to the other side so it is multiplied not divided. This removes the risk of divide by zero while maintaining a conservative reference. If these checks fail then the land_complete flag will be used to move to the next step.

**Comment by lthall** [comment]
>> This feature is rarely used, if this has had some testing I would be inclined to merge.

We use this feature a lot with both slung and hard mounted payloads on the Callisto. This is where we have found the problem of early touch down detection and disarming on landing.

I also noted that on landing payload place stops running so I have also let it continue if it is not landed at the start of payload place.

I will get the Iris out to show some results.

**Comment by lthall** [comment]
image

Testing using the Iris. The measurement, touchdown, detection, position controller reset, take-off and continuation of the mission is all shown and working as expected.

**Comment by lthall** [comment]
I am happy for this to go in when checks pass.

**Pull request merged**